qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/general_fitting/general_fitting_model.py

```diff
@@ -218,6 +218,9 @@ def _get_new_domain_function_for(self, new_dataset_name: str) -> IFunction:
 
     def _add_global_ties_to_simultaneous_function(self) -> None:
         """Creates and adds ties to the simultaneous function to represent the global parameters."""
+        assert self.simultaneous_fit_function is not None, "This method assumes the simultaneous fit function is not None."
```

Comment:
Is this assert executed in the release? What is the advantage over throwing an error?

Reply:
Its only executed in debug builds, so not in release

Its an example of defensive programming where its generally considered good practise to assert on the assumptions you  make for the inputs of a function. This is also good for self-documenting reasons

The problem with throwing an error is that the user will see this error, even when they haven't done anything wrong and can't do anything about it. I have started using assertions to indicate when there is a logical problem in the developers code, rather than it being down to user error

Reply:
Thanks for the explanation - I'll start using them!